### PR TITLE
Feature get course by id swen900132021 ag 95

### PR DESCRIPTION
**Pull request by Akameea**

### CODE REVIEW

File: src/routes/courses.js [hunk 1]
```diff
@@ -16,7 +16,11 @@ router.get('/:course_id', (req, res) => {
     res.json(courseloader.getCourseById(req.params.course_id));
   } else {
     res.status(404);
-    res.send('course_id does not exist!');
+    res.json({
+      status: 404,
+      message: 'Failed to find the course ID',
+      reason: 'Course ID does not exist',
```
Comment (Ev-anz) [inline src/routes/courses.js:22]: 'reason' is not really needed, its content can be included in message as well, we want to make the message returned to frontend as clear as possible

**Review by Ev-anz — CHANGES_REQUESTED** [review]

File: src/routes/courses.js (hunk 1 above)
Comment (Akameea) [reply src/routes/courses.js:22]: Sure.

**Review by Ev-anz — APPROVED** [review]

File: src/routes/courses.js [hunk 2]
```diff
@@ -16,7 +16,10 @@ router.get('/:course_id', (req, res) => {
     res.json(courseloader.getCourseById(req.params.course_id));
   } else {
     res.status(404);
```
Comment (Ev-anz) [inline src/routes/courses.js:18]: Just realized there's a duplicate status 404......Stupid of me not noticing this on first sight, maybe remove the res.status(404) statement

Comment (junkaixing) [reply src/routes/courses.js:18]: what do you mean? Why don't we need to specify the return status when things go wrong? 404 represent page not found. Shall we use 406 in this case to indicate that if there is no such course found with the given course ID from the client?

**Review by Ev-anz — APPROVED** [review]

**Pull request merged**